spring-cloud-gcp-samples/spring-cloud-gcp-trace-sample/src/test/java/com/example/TraceSampleApplicationIntegrationTests.java

```diff
@@ -82,17 +95,42 @@
 
   private TraceServiceBlockingStub traceServiceStub;
 
-  @BeforeClass
-  public static void checkToRun() {
-    assumeThat(
-        "Google Cloud Trace integration tests are disabled. "
-            + "Please use '-Dit.trace=true' to enable them. ",
-        System.getProperty("it.trace"),
-        is("true"));
+  static final String projectName =
+      ProjectName.of(ServiceOptions.getDefaultProjectId()).getProject();
+
+  private static TopicAdminClient topicAdminClient;
+
+  private  static SubscriptionAdminClient subscriptionAdminClient;
+
+  private static PubSubAdmin pubSubAdmin;
+
+  @BeforeAll
+  static void setup() throws IOException {
+
+    projectIdProvider = () -> projectName;
```

Comment:
Since `projectName` is only used once, you can directly use `ServiceOptions.getDefaultProjectId()` here without an intermediate constant.